packages/pangea-sdk/pangea/asyncio/services/vault.py

```diff
@@ -321,719 +329,1422 @@ async def update(
-
-        Import a secret
+        Store secret
 
-        OperationId: vault_post_v1_secret_store 1
+        Store a Pangea token.
 
         Args:
-            secret (str): The secret value
-            name (str): The name of this item
-            folder (str, optional): The folder where this item is stored
-            metadata (dict, optional): User-provided metadata
-            tags (list[str], optional): A list of user-defined tags
-            rotation_frequency (str, optional): Period of time between item rotations
-            rotation_state (ItemVersionState, optional): State to which the previous version should transition upon rotation.
-                Supported options:
-                - `deactivated`
-                - `destroyed`
-            expiration (str, optional): Expiration timestamp
+            token: The Pangea token value.
+            name: The name of this item.
+            folder: The folder where this item is stored.
+            metadata: User-provided metadata.
+            tags: A list of user-defined tags.
+            disabled_at: Timestamp indicating when the item will be disabled.
 
         Raises:
             PangeaAPIException: If an API Error happens
 
-        Returns:
-            A PangeaResponse where the secret
-                is returned in the response.result field.
-                Available response fields can be found in our [API documentation](https://pangea.cloud/docs/api/vault#import-a-secret).
-
         Examples:
-            response = vault.secret_store(
-                secret="12sdfgs4543qv@#%$casd",
-                name="my-very-secret-secret",
-                folder="/personal",
-                metadata={
-                    "created_by": "John Doe",
-                    "used_in": "Google products"
-                },
-                tags=[
-                    "irs_2023",
-                    "personal"
-                ],
-                rotation_frequency="10d",
-                rotation_state=ItemVersionState.DEACTIVATED,
-                expiration="2025-01-01T10:00:00Z",
-            )
+            response = await vault.store_pangea_token(token="foobar")
         """
-        input = SecretStoreRequest(
-            type=ItemType.SECRET,
-            secret=secret,
+
+        return await self._secret_store(
+            item_type="pangea_token",
+            result_class=PangeaToken,
+            token=token,
             name=name,
             folder=folder,
             metadata=metadata,
             tags=tags,
+            disabled_at=disabled_at,
             rotation_frequency=rotation_frequency,
             rotation_state=rotation_state,
-            expiration=expiration,
+            rotation_grace_period=rotation_grace_period,
         )
-        return await self.request.post("v1/secret/store", SecretStoreResult, data=input.model_dump(exclude_none=True))
 
-    async def pangea_token_store(
+    async def store_pangea_client_secret(
```

Comment:
Wait, why you overloaded key_store() and not this one?

Reply:
This one is harder to cleanly overload because Pangea client secret takes 3 input args but secret and Pangea token take only one. And even then, we would need to ask the user to specify an `item_type` in order to differentiate between the types, and at that point it's almost no different from having different named functions. At least with `store_key()` the overloads are combinations across 3+ args (`key_type`, `purpose`, `algorithm`) and help to ensure that e.g. only asymmetric algorithms are specified for asymmetric keys.